extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesApplyReplicasDecorator.java

```diff
@@ -0,0 +1,22 @@
+package io.quarkus.kubernetes.deployment;
+
+import io.dekorate.kubernetes.decorator.ApplyReplicasDecorator;
+import io.dekorate.kubernetes.decorator.Decorator;
+import io.dekorate.kubernetes.decorator.NamedResourceDecorator;
+
+public class KubernetesApplyReplicasDecorator extends ApplyReplicasDecorator {
+
+    public KubernetesApplyReplicasDecorator(int replicas) {
+        super(replicas);
+    }
+
+    public KubernetesApplyReplicasDecorator(String deploymentName, int replicas) {
+        super(deploymentName, replicas);
+    }
+
+    @Override
+    public Class<? extends Decorator>[] after() {
+        return new Class[] { NamedResourceDecorator.class };
```

Comment:
Just to be on the safe side of things, I would explicitly express that I want to it to be called after the ApplyReplicasDecorator. Why?

1. It justifies the purpose of the class
2. It makes it easy to understand which dekorator is applied first. (e.g. atm I can't understand which will be applied first without checking the sources of AppliedReplicasDecorator).

Reply:
Hi @iocanel this was my initial intension and I had implemented it but failed. I now changed it as your said and it seems to be working. I think I had a package problem.

My intension was to enable this decorator to work after all NamedResourceDecorators. However, the method you suggested is much more understandable.

Reply:
I change the after call to ApplyReplicasDecorator